File: src/main/java/com/ggardet/patient/internal/repository/MedicalRecordRepository.java
package com.ggardet.patient.internal.repository;

import com.ggardet.patient.internal.entity.MedicalRecord;
import org.springframework.data.jpa.repository.JpaRepository;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

public interface MedicalRecordRepository extends JpaRepository<MedicalRecord, UUID> {
    List<MedicalRecord> findByPatientId(final UUID patientId);
    List<MedicalRecord> findByDoctorId(final UUID doctorId);
    List<MedicalRecord> findByRecordDateBetween(final LocalDateTime start, final LocalDateTime end);
    List<MedicalRecord> findByFollowUpRequired(final Boolean followUpRequired);
}

File: src/main/kotlin/com/ggardet/patient/internal/repository/MedicalRecordRepository.kt
package com.ggardet.patient.internal.repository

import com.ggardet.patient.internal.entity.MedicalRecord
import org.springframework.data.jpa.repository.JpaRepository
import java.time.LocalDateTime
import java.util.*

interface MedicalRecordRepository : JpaRepository<MedicalRecord, UUID> {
    fun findByPatientId(patientId: UUID): List<MedicalRecord>
    fun findByDoctorId(doctorId: UUID): List<MedicalRecord>
    fun findByRecordDateBetween(start: LocalDateTime, end: LocalDateTime): List<MedicalRecord>
    fun findByFollowUpRequired(followUpRequired: Boolean): List<MedicalRecord>
}